pkg/autoscaler/aggregation/bucketing.go

```diff
@@ -87,32 +87,20 @@ func (t *TimedFloat64Buckets) RemoveOlderThan(time time.Time) {
 	}
 }
 
-// float64Bucket keeps all the stats that fall into a defined bucket.
-type float64Bucket map[string]float64Value
-
 // float64Value is a single value for a Float64Bucket. It maintains a summed
 // up value and a count to ultimately calculate an average.
 type float64Value struct {
 	sum   float64
 	count float64
 }
 
-// record adds a value to the bucket. Buckets with the same given name
-// will be collapsed.
-func (b float64Bucket) record(name string, value float64) {
-	current := b[name]
-	b[name] = float64Value{
-		sum:   current.sum + value,
-		count: current.count + 1.0,
+func (b *float64Value) avg() float64 {
+	if b.count == 0 {
+		return 0
 	}
+	return b.sum / b.count
```

Comment:
If I understand correctly, `b.sum` is a sum of individual averages. If that is true, then this computation is the average-of-averages, which is actually [not a very accurate or useful measure](https://math.stackexchange.com/questions/95909/why-is-an-average-of-an-average-usually-incorrect).

It would be an improvement to this calculation to make it weighted averages instead of average-of-averages, if possible.

Reply:
there's a long history why we were doing this. You can watch today's ASWG meeting for some context :-)